pkg/controller/load_configuration.go

```diff
@@ -0,0 +1,96 @@
+package controller
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+// LoadFromSecret retrieves an operator secret
+func LoadFromSecret(prefix, resourceKey string, cl client.Client) error {
+	// get the secret name
+	secretName := getResourceName(resourceKey)
+	if secretName == "" {
+		return nil
+	}
+
+	// get the secret
+	namespace, err := k8sutil.GetWatchNamespace()
+	if err != nil {
+		return err
+	}
+
+	secret := &v1.Secret{}
+	namespacedName := types.NamespacedName{Namespace: namespace, Name: secretName}
+	err = cl.Get(context.TODO(), namespacedName, secret)
+	if err != nil {
+		return err
```

Comment:
We should ignore NotFound errors. But we want to log a message saying that there is no Secret so we don't load anything from it.

Reply:
this is done in each operator, but i guess i can move it here, i just thought it would be better to have it in the operator so we have more control over the error at that point

Reply:
done here: 26869a4875e8152516ff11b763471d23defe1670